tests/unit/ad/test_ad_operator_tree.py

```diff
@@ -257,13 +257,30 @@ def test_time_dependent_array():
             name="bar", values=vals_it, data=intf_data, iterate_index=0
         )
 
+    for bg, bg_data in mdg.boundaries(return_data=True):
+        vals_sol = np.arange(bg.num_cells)
+        pp.set_solution_values(
+            name="foobar", values=vals_sol, data=bg_data, time_step_index=0
+        )
+
+        vals_it = np.ones(bg.num_cells)
+        pp.set_solution_values(
+            name="foobar", values=vals_sol, data=bg_data, iterate_index=0
+        )
+
     # We make three arrays: One defined on a single subdomain, one on all subdomains of
     # mdg and one on an interface.
     sd_array_top = pp.ad.TimeDependentDenseArray(
-        "foo", subdomains=mdg.subdomains(dim=mdg.dim_max())
+        "foo", domains=mdg.subdomains(dim=mdg.dim_max())
     )
-    sd_array = pp.ad.TimeDependentDenseArray("foo", subdomains=mdg.subdomains())
-    intf_array = pp.ad.TimeDependentDenseArray("bar", interfaces=mdg.interfaces())
+    sd_array = pp.ad.TimeDependentDenseArray("foo", domains=mdg.subdomains())
+    intf_array = pp.ad.TimeDependentDenseArray("bar", domains=mdg.interfaces())
+    bg_array = pp.ad.TimeDependentDenseArray("foobar", domains=mdg.boundaries())
+
+    # Check correct domain types
+    assert sd_array.domain_type == sd_array_top.domain_type == "subdomains"
+    assert intf_array.domain_type == "interfaces"
+    assert bg_array.domain_type == "boundary grids"
```

Comment:
Should we not also test evaluation of bg_array?

Reply:
Included.